### PR TITLE
create RelatedProductList with list.context 

### PR DESCRIPTION
**Pull request by Luis-eplus**
#### What problem is this solving?

Make the use of related products shelf more consistent with regular shelf

#### How to test it?

Add `list-context.related-product-list` in your store theme, with the `slider-layout` as children and `product-summary.shelf` as block

[Workspace](https://luis--lojasupersaudavel.myvtex.com/)

#### Screenshots or example usage:

image

#### How does this PR make you feel? [:link:](http://giphy.com/)

)

### CODE REVIEW

File: messages/context.json [hunk 1]
```diff
@@ -70,5 +70,16 @@
   "admin/editor.productSummaryList.installmentCriteria.max-without-interest": "Maximum without interest enum name",
   "admin/editor.productSummaryList.installmentCriteria.max-with-interest": "Maximum with interest enum name",
   "admin/editor.productSummaryList.analyticsListName.title": "List name in Analytics",
-  "admin/editor.productSummary.trackListName.title": "Should track list name"
+  "admin/editor.productSummary.trackListName.title": "Should track list name",
```
Comment (sheilagomes) [inline messages/context.json:73]:
```suggestion
  "admin/editor.productSummary.trackListName.title": "Should track list name",
  >>>>> "admin/editor.productSummary.trackListName.title": "Track list name",
```

File: messages/en.json [hunk 2]
```diff
@@ -70,5 +70,16 @@
   "admin/editor.productSummaryList.installmentCriteria.max-without-interest": "Maximum without interest",
   "admin/editor.productSummaryList.installmentCriteria.max-with-interest": "Maximum",
   "admin/editor.productSummaryList.analyticsListName.title": "List name in Analytics",
-  "admin/editor.productSummary.trackListName.title": "Should track list name"
+  "admin/editor.productSummary.trackListName.title": "Should track list name",
```
Comment (sheilagomes) [inline messages/en.json:73]:
```suggestion
  "admin/editor.productSummary.trackListName.title": "Should track list name",
```  >>>>> "admin/editor.productSummary.trackListName.title": "Track list name",

Comment (sheilagomes) [inline messages/en.json:73]:
```suggestion
  "admin/editor.productSummary.trackListName.title": "Track list name",
```

File: messages/en.json [hunk 3]
```diff
@@ -70,5 +70,16 @@
...
+  "admin/editor.productSummaryRelatedList.title": "Related Products",
+  "admin/editor.productSummaryRelatedList.description": "A related products shelf to show view recommendations",
```
Comment (sheilagomes) [inline messages/en.json:75]:
```suggestion
  "admin/editor.productSummaryRelatedList.description": "A window display showing product recommendations for the user to view",
```

File: messages/en.json [hunk 4]
```diff
@@ -70,5 +70,16 @@
...
+  "admin/editor.productSummaryRelatedList.recommendation": "Recommendations",
+  "admin/editor.productSummaryRelatedList.recommendation.description": "Which recommendations will be shown",
```
Comment (sheilagomes) [inline messages/en.json:77]:
```suggestion
  "admin/editor.productSummaryRelatedList.recommendation.description": "Recommendations that will be shown",
```

File: messages/en.json [hunk 5]
```diff
@@ -70,5 +70,16 @@
...
+  "admin/editor.productSummaryRelatedList.similars": "Similar products",
+  "admin/editor.productSummaryRelatedList.view": "Who saw this, also saw...",
```
Comment (sheilagomes) [inline messages/en.json:79]:
```suggestion
  "admin/editor.productSummaryRelatedList.view": "People who viewed this also viewed...",
```

File: messages/en.json [hunk 6]
```diff
@@ -70,5 +70,16 @@
...
+  "admin/editor.productSummaryRelatedList.buy": "Who bought this, also bought...",
+  "admin/editor.productSummaryRelatedList.accessories": "Accessories products",
+  "admin/editor.productSummaryRelatedList.viewAndBought": "Who saw this, also bought...",
```
Comment (sheilagomes) [inline messages/en.json:82]:
```suggestion
  "admin/editor.productSummaryRelatedList.accessories": "Accessories",
  "admin/editor.productSummaryRelatedList.viewAndBought": "People who saw this, also bought...",
```

File: messages/es.json [hunk 7]
```diff
@@ -70,5 +70,16 @@
   "admin/editor.productSummaryList.installmentCriteria.max-without-interest": "Máximo sin interés",
   "admin/editor.productSummaryList.installmentCriteria.max-with-interest": "Máximo",
   "admin/editor.productSummaryList.analyticsListName.title": "Nombre de la lista en Analytics",
-  "admin/editor.productSummary.trackListName.title": "Debe rastrear el nombre de la lista"
+  "admin/editor.productSummary.trackListName.title": "Debe rastrear el nombre de la lista",
+  "admin/editor.productSummaryRelatedList.title": "Productos Relacionados",
+  "admin/editor.productSummaryRelatedList.description": "Una vitrina de productos relacionados para mostrar recomendaciones de vista",
```
Comment (sheilagomes) [inline messages/es.json:75]:
```suggestion
  "admin/editor.productSummary.trackListName.title": "Rastrear el nombre de la lista",
  "admin/editor.productSummaryRelatedList.title": "Productos relacionados",
  "admin/editor.productSummaryRelatedList.description": "Una vitrina de productos recomendados para ver",
```

File: messages/es.json [hunk 8]
```diff
@@ -70,5 +70,16 @@
...
+  "admin/editor.productSummaryRelatedList.recommendation": "Recomendaciones",
+  "admin/editor.productSummaryRelatedList.recommendation.description": "Qué recomendaciones se mostrará",
```
Comment (sheilagomes) [inline messages/es.json:77]:
```suggestion
  "admin/editor.productSummaryRelatedList.recommendation.description": "Recomendaciones que se mostrarán",
```

File: messages/pt-BR.json [hunk 9]
```diff
@@ -70,5 +70,16 @@
   "admin/editor.productSummaryList.installmentCriteria.max-without-interest": "Máxima sem juros",
   "admin/editor.productSummaryList.installmentCriteria.max-with-interest": "Máxima",
   "admin/editor.productSummaryList.analyticsListName.title": "Nome da lista no Analytics",
-  "admin/editor.productSummary.trackListName.title": "Deveria rastrear nome da lista"
+  "admin/editor.productSummary.trackListName.title": "Deveria rastrear nome da lista",
+  "admin/editor.productSummaryRelatedList.title": "Produtos Relacionados",
```
Comment (sheilagomes) [inline messages/pt-BR.json:74]:
```suggestion
  "admin/editor.productSummaryRelatedList.title": "Produtos relacionados",
```

File: messages/pt.json [hunk 10]
```diff
@@ -70,5 +70,16 @@
   "admin/editor.productSummaryList.installmentCriteria.max-without-interest": "Máxima sem juros",
   "admin/editor.productSummaryList.installmentCriteria.max-with-interest": "Máxima",
   "admin/editor.productSummaryList.analyticsListName.title": "Nome da lista no Analytics",
-  "admin/editor.productSummary.trackListName.title": "Deveria rastrear nome da lista"
+  "admin/editor.productSummary.trackListName.title": "Deveria rastrear nome da lista",
+  "admin/editor.productSummaryRelatedList.title": "Produtos Relacionados",
```
Comment (sheilagomes) [inline messages/pt.json:74]:
```suggestion
  "admin/editor.productSummaryRelatedList.title": "Produtos relacionados",
```

**Review by sheilagomes — CHANGES_REQUESTED** [review]

**Review by sheilagomes — APPROVED** [review]
LGTM

**Comment by victorgrodriguesm7** [comment]
Any ETA for this being merged ? I started a new project where this Shelf is needed. Since the ProductSummary does not have this type of Shelf and the old Lib(`vtex.shelf`) is deprecated, I need to know if this merge will happen soon.

**Pull request closed without merging**